.github/workflows/code-coverage.yml

```diff
@@ -0,0 +1,138 @@
+name: 'Code Coverage'
+on:
+  workflow_call:
+    inputs:
+      type:
+        description: Type of report (unit or integration)
+        type: string
+        default: unit
+      resource_name:
+        description: Resource name of report folder
+        type: string
+
+jobs:
+  coverage-unit:
+    runs-on: ubuntu-latest
+    name: Jest tests coverage
+    if: ${{ inputs.type == 'unit' }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Download Coverage Report
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ inputs.resource_name }}
+          path: report
+
+      - uses: jwalton/gh-find-current-pr@v1
+        id: findPr
+
+      - uses: ArtiomTr/jest-coverage-report-action@v2
+        with:
+          prnumber: ${{ steps.findPr.outputs.number }}
+          coverage-file: report/coverage/report.json
+          base-coverage-file: report/coverage/report.json
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          skip-step: all
+          custom-title: Code Coverage - ${{ inputs.resource_name == 'report-be' && 'Backend' || 'Frontend' }} unit tests
+
+  coverage-integration:
+    runs-on: ubuntu-latest
+    name: Jest tests coverage
+    if: ${{ inputs.type == 'integration' }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Download Coverage Report
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ inputs.resource_name }}
+
+      - name: Parse Coverage Summary
+        id: parse-coverage
+        run: |
+          # Define a reusable function to extract coverage data
+          extract_coverage_data() {
+              local keyword=$1
+              local field=$2
+              awk "/$keyword/ {print $field}" integration-coverage.txt | tr -d '\n|%'
+          }
+
+          # Determine status based on percentage
+          get_status() {
+              if [ "$(echo "$1 < 50" | bc)" -eq 1 ]; then
+                  echo "🔴"
+              elif [ "$(echo "$1 < 80" | bc)" -eq 1 ]; then
+                  echo "🟡"
+              else
+                  echo "🟢"
+              fi
+          }
+
+          # Extract coverage data from the summary
+          STATEMENTS_PERCENT=$(extract_coverage_data "Statements" '$3')
```

Comment:
Let's add a comment above to help with the placeholder stuff:
Example: Statements : 78.90% (123/156)
$1 -> Statements
$2 -> :
$3 ...